src/Truncate/index.jsx

```diff
@@ -0,0 +1,57 @@
+import React, {
+  useEffect, useRef, useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import { truncateLines } from './utils';
+import { useWindowSize } from '../index';
+
+const DEFAULT_TRUNCATE_LINES = 1;
+const DEFAULT_TRUNCATE_ELLIPSIS = '...';
+const DEFAULT_TRUNCATE_ELEMENT_TYPE = 'div';
+
+const Truncate = ({
+  children, lines, ellipsis, elementType, className, whiteSpace,
+}) => {
+  const [truncateText, setTruncateText] = useState('');
+  const textContainer = useRef();
+  const { width } = useWindowSize();
+
+  useEffect(() => {
```

Comment:
we may want to use `useLayoutEffect` here, as its generally recommended when we're working with DOM mutations to provide a smoother experience.

Reply:
It`s good idea, thanks